src/Toolkit/Toolkit.Maui/BasemapGallery/BasemapGallery.cs

```diff
@@ -41,9 +41,17 @@ public BasemapGallery()
         ListItemTemplate = DefaultListDataTemplate;
         GridItemTemplate = DefaultGridDataTemplate;
         ControlTemplate = DefaultControlTemplate;
-        _ = _controller.LoadFromDefaultPortal();
-    }
-
+        Loaded += BasemapGallery_Loaded;
+    }
+
+    private void BasemapGallery_Loaded(object? sender, EventArgs e)
+    {
+        if (AvailableBasemaps is null)
+        {
+            _ = _controller.LoadFromDefaultPortal();

```

Comment:
We might need to consider making this cancellable in case AvailableBasemaps does get set after this code was called.